app/connectors/ApiScopeConnector.scala

```diff
@@ -20,27 +20,28 @@ import config.AppConfig
 import javax.inject.{Inject, Singleton}
 import model._
 import model.Environment.Environment
-import uk.gov.hmrc.http.{HeaderCarrier, Upstream5xxResponse}
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.http.HttpReads.Implicits._
 
 import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.http.HttpErrorFunctions.is5xx
 
 abstract class ApiScopeConnector(implicit ec: ExecutionContext) {
   protected val httpClient: HttpClient
-  protected val proxiedHttpClient: ProxiedHttpClient
   val environment: Environment
   val serviceBaseUrl: String
-  val useProxy: Boolean
-  val bearerToken: String
-  val apiKey: String
 
-  def http: HttpClient = if (useProxy) proxiedHttpClient.withHeaders(bearerToken, apiKey) else httpClient
+  def http: HttpClient
+
+  private def for5xx(ex: Throwable): PartialFunction[Throwable,Nothing] = (err: Throwable) => err match {
+    case e: UpstreamErrorResponse if(is5xx(e.statusCode)) => throw ex 
+  }
 
   def fetchAll()(implicit hc: HeaderCarrier): Future[Seq[ApiScope]] = {
     http.GET[Seq[ApiScope]](s"$serviceBaseUrl/scope")
-      .recover {
-        case _: Upstream5xxResponse => throw new FetchApiDefinitionsFailed
-      }
+    .recover(for5xx(new FetchApiDefinitionsFailed)) // TODO - odd choice of exception
```

Comment:
does anything need doing RE the "ToDo"?

Reply:
We have looked at the TODOs and none of them can be done just yet